lib/internal/mime.js

```diff
@@ -80,43 +158,76 @@ function parseTypeAndSubtype(str) {
   };
 }
 
-const EQUALS_SEMICOLON_OR_END = /[;=]|$/;
-const QUOTED_VALUE_PATTERN = /^(?:([\\]$)|[\\][\s\S]|[^"])*(?:(")|$)/u;
-
-function removeBackslashes(str) {
-  let ret = '';
-  // We stop at str.length - 1 because we want to look ahead one character.
-  let i;
-  for (i = 0; i < str.length - 1; i++) {
-    const c = str[i];
-    if (c === '\\') {
-      i++;
-      ret += str[i];
+function isSemicolonOrEquals(char) {
+  const code = StringPrototypeCharCodeAt(char, 0);
+  return code === SEMICOLON_CHARCODE || code === EQUALS_CHARCODE;
+}
+
+function moveUntil(str, predicate, marker, {
+  toASCII,
+  removeBackslashes,
+} = {
+  toASCII: false,
+  removeBackslashes: false,
+},
+) {
+  const start = marker.position;
+  const length = str.length;
+  let index = start;
+  let result = '';
+  let char = str[index];
+
+  while (index < length && predicate(char) === false) {
+    let asCharCode;
+    let controlCharCode;
+    // We lazily assign it if needed, as not all the iterations
+    // might require it
+    if (removeBackslashes && (
+      asCharCode = StringPrototypeCharCodeAt(char, 0)
+    ) === BACKSLASH_CHARCODE
+    ) {
+      // Jump to the next char
+      index++;
+      char = str[index];
+      controlCharCode = StringPrototypeCharCodeAt(char, 0);
+    }
+
+    if (toASCII) {
+      asCharCode = controlCharCode != null ?
+        controlCharCode :
+        asCharCode || StringPrototypeCharCodeAt(char, 0);
+      result += asCharCode >= ASCII_LOWER_LOWERBOUND &&
+        asCharCode <= ASCII_LOWER_UPPERBOUND ?
+        StringPrototypeToLowerCase(char) :
+        char;
     } else {
-      ret += c;
+      result += char;
     }
+
+    index++;
+    char = str[index];
   }
-  // We add the last character if we didn't skip to it.
-  if (i === str.length - 1) {
-    ret += str[i];
-  }
-  return ret;
-}
 
+  marker.position = index;
+
+  return result;
+}
 
 function escapeQuoteOrSolidus(str) {
   let result = '';
+
   for (let i = 0; i < str.length; i++) {
     const char = str[i];
-    result += (char === '"' || char === '\\') ? `\\${char}` : char;
+    const code = StringPrototypeCharCodeAt(char, 0);
+    result += (code === QUOTE_CHARCODE || char === BACKSLASH_CHARCODE) ? `\\${char}` : char;
```

Comment:
maybe branchless is faster

```suggestion
    result += (code === QUOTE_CHARCODE || char === BACKSLASH_CHARCODE) && `\\${char}` || char;
```

What if we define an Array [QUOTE_CHARCODE, BACKSLASH_CHARCODE] and use indexOf instead of two compares?

Reply:
Strict equals seems faster on all combination, but this is merely empirical, does not contemplates the current use cases for the PR:

```shell
Array#indexOf x 111,610,492 ops/sec ±0.10% (197 runs sampled)
Array#includes x 111,627,347 ops/sec ±0.07% (197 runs sampled)
Strict x 1,030,052,905 ops/sec ±0.46% (197 runs sampled)
Fastest is Strict
```

```js
const Benchmark = require('benchmark');

const suite = new Benchmark.Suite();

const array = [1, 5]
const valueA = 5;
const valueB = 1;
const valueC = 3;

// Tested different combinations, not only the ones showed here
suite
  .add('Array#indexOf', function () {
    return (array.indexOf(valueC) || array.indexOf(valueA + 1) || array.indexOf(valueB)) > 0
  }, { minSamples: 100 })
  .add('Array#includes', function () {
    return (array.indexOf(valueC) || array.indexOf(valueA + 1) || array.indexOf(valueB)) > 0
  }, { minSamples: 100 })
  .add('Strict', function () {
    return valueC === 5 || valueA === 5 + 1 || valueB === 1
  }, { minSamples: 100 })
  .on('cycle', function (event) {
    console.log(String(event.target));
  })
  .on('complete', function () {
    console.log('Fastest is ' + this.filter('fastest').map('name'));
  })
  .run();
```